src/io/export_stl.cc

```diff
@@ -48,12 +50,19 @@ Vector3d toVector(const std::array<double, 3>& pt) {
   return {pt[0], pt[1], pt[2]};
 }
 
-Vector3d fromString(const std::string& vertexString)
+Vector3f fromString(const std::string& vertexString)
 {
-  Vector3d v;
-  std::istringstream stream{vertexString};
-  stream >> v[0] >> v[1] >> v[2];
-  return v;
+  auto cstr = vertexString.c_str();
+  return {
+    strtof(cstr, (char**)&cstr), 
```

Comment:
Does that work in locales having something else than '.' as decimal separator? The man page says it's using locale settings.

Reply:
Ah! Luckily there's a a `setlocale(LC_NUMERIC, "C");` in the parent of all append_stl functions that's meant to force '.' as decimal separator for both ways of parsing in this context (weirdly tho, I tried to force the locale to "fr_FR.UTF-8" there and the output didn't budge - I was expecting commas 🧐)

Reply:
Keep in mind that `setLocale` has been shown to be incredibly slow on windows if you enable localization in preferences. #2744

Reply:
@thehans oh, does that setlocale currently make ASCII STL export slower on Windows? (sounds like something exciting to follow up on, although this PR shouldn't make things worse in that respect)

Reply:
I don't know how much it is specifically affecting STL output speed.  
IIRC the test case there was particularly slow during CSG generation (sample code had tons of polyhedron data, thousands of vertex coordinates that require double to string conversion).

That was the impetus for adding `double-conversion` library, and so any time we print a number through our `Value` variant class it now goes through that.  But I don't think any of the export code was ever touched as part of those changes.

There has been some discussion more recently about switching that out for `libfmt` aka {fmt} which is supposed to be even faster than `double-conversion`.
I spent quite some time a few months ago attempting to swap out the libraries but didn't quite get it to a fully working/presentable state.  I might revisit it soon if I find the energy.

Reply:
It might not be particularly relevant in this case, I had a bit of a knee-jerk reaction to seeing the mention of `setlocale`.  But as long as its not being called on every number conversion, it shouldn't be too terrible.  
The problem was that we weren't even calling setlocale directly, it was just a consequence of the code path taken deep within `printf` or  `ostream operator<<` or whatever the previous code was calling.

Reply:
Also I think the latest windows builds should be linking with `ucrt` instead of `msvcrt`, which might not be affected by those same performance issues.

Reply:
@thehans Thanks for the pointers! Also, caveat: been a bit feverish this week so don't take anything I write at face value 🤒🤪 (trying to find the [ideal body temperature for coding](https://xkcd.com/323/))

I've tested double-conversion ([source here](https://gist.github.com/ochafik/fca4d509b9e8cd314c7fe49153724f2b)) and it seems faster than STR (ostringstream), but not as fast as a simple `sprintf(output, "%.6g %.6g %.6g", v[0], v[1], v[2]);`. In the other direction, it's also slower than `strtof`.

I've now also fallen into the rabbit hole of https://github.com/openscad/openscad/issues/2651 (now hoping to switch from 6 to 9 digits in this PR - might still be faster btw), and realized this all needs better test cases, which I'll add before further experiments (I've probably already silently broken the round-trip logic for distinct vertices that are formatted to the same string, for instance). Good thing this is still officially a draft 😅

Reply:
You might also try casting `double` to `float` and calling [`ToShortestSingle`](https://github.com/google/double-conversion/blob/master/double-conversion/double-to-string.h#LL245C8-L245C24)

Reply:
Re/ doubleconv it’s weird indeed, maybe I’ve misused it (tried same Params as the Value code, then more naive params), or the mac stdlib is faster than average (might try [this benchmark](https://github.com/miloyip/dtoa-benchmark) if I have a chance), or… the way I naively stripped trailing zeros negated its performance? (Hadn’t seen  ToShortestSingle, will try next)

Reply:
From your gist `trimTrailingZeros`  doesn't appear to actually get called anywhere.  

Anyways, simply stripping zeroes from the end is not even remotely correct behavior, in the event that:
1) There is no decimal point:  "1000" becomes "1"
2) There is an exponent
   a) "1.000000e6" would not get trimmed at all  
   b) "1.000000e10" would trim the exponent "1.000000e1" #2950

Reply:
The more I think about it, I feel that using `ToShortestSingle` would make the most sense here, and would allow for simplifying the code much further:

`ToShortestSingle` creates the shortest **exact representation** of the input value (a "round trip" guarantee).  So no need to worry about trailing zeros (and no directly specifying any precision)

Additionally, all of the code going from double to string and back, was for the purpose of normal calculations matching up after reading in the decimal ASCII converted vertex positions (#1853).
If you downcast `double` to `float` and calculate normals from that, then there's no need to perform the round trip (string back to float) ourselves, because no further precision will be lost(post-downcast) by `ToShortestSingle`.